parse_node/parse_source_file.ts

```diff
@@ -21,18 +28,78 @@ const getClassDeclarationHeader = (
 
     const clause = node.heritageClauses[0] as ts.HeritageClause
     const type = clause.types[0]
+    const tsType = props.program.getTypeChecker().getTypeAtLocation(type)
 
     extendsFrom = type.getText()
+
+    if (tsType.symbol && tsType.symbol.declarations) {
```

Comment:
Would it actually be helpful to build a map of "class" to "is inner or normal" as a pass before codegen? And then we could just pass that into all the nodes in ParseState?

Mostly just asking out of curiosity, I can't really tell.

Reply:
I think one advantage of this is it'll help you reason about the whole "single class in a file is automatically default" case that I was suggesting without having to put that if statement everywhere... :)

Reply:
I don't know if "single class in a file is automatically default" is a good idea because what if someone wants a `.gd` only with one inner class?  (asked the same question in my [comment below](https://github.com/johnfn/ts2gd/pull/61#issuecomment-997855700)). 

> Would it actually be helpful to build a map of "class" to "is inner or normal" as a pass before codegen? And then we could just pass that into all the nodes in ParseState?

About that I think we could improve it in the future but currently there is no need to do this. I don't want to make this PR even more complicated :)